src/main/java/com/hotels/bdp/circustrain/bigquery/partition/BigQueryPartitionGenerator.java

```diff
@@ -93,9 +96,13 @@ private String getQueryStatement() {
     return part;
   }
 
-  private ExtractionUri scheduleForExtraction(com.google.cloud.bigquery.Table table) {
+  private ExtractionUri scheduleForExtraction(com.google.cloud.bigquery.Table table, Partition partition) {
     ExtractionUri extractionUri = new ExtractionUri(destinationBucket, generateFolderName(), generateFileName());
-    ExtractionContainer toRegister = new ExtractionContainer(table, extractionUri, PostExtractionAction.DELETE);
+    PostExtractionAction deleteAction = new DeletePostExtractionAction(table);
+    PostExtractionAction updatePartitionSchemaAction = new UpdatePartitionSchemaPostExtractionAction(partition,
+        extractionService.getStorage(), extractionUri);
+    ExtractionContainer toRegister = new ExtractionContainer(table, extractionUri,
+        Arrays.asList(deleteAction, updatePartitionSchemaAction));
```

Comment:
Minor, and it probably doesn't make a difference at the moment, but shouldn't the order here be `updatePartitionSchemaAction` and then `deleteAction`?

Reply:
In `deleteAction`, the BigQuery table is deleted, whereas `updatePartitionSchemaAction` uses the exported avro file that is in Google Cloud Storage to get the schema. They don't depend on each other, so the order shouldn't matter :)